generate-repo-cache.nu

```diff
@@ -12,8 +12,43 @@ def get-metadata [
     |from yaml
 }
 
+def check-required-attributes [
+    metadata
+] {
+    $metadata
+    |each { |entry|
+        ["author" "name" "os" "short-desc" "url" "version" "keywords"]
+        |each {|attribute|
+            if ($attribute not-in $entry) {
+                error make {msg: $"$($entry) lacks: $($attribute)"}
+                exit 1
+            }
+        }
+    }
+}
+
+def add-optional-attributes [
+    metadata
+] {
+    if ("name" not-in $metadata) {
+                error make {msg: $"$($metadata) lacks: name"}
+                exit 1
+    }
+
+    {"pre-install-msg": "",
```

Comment:
this should be moved to it's own variable (or function returning it since we still don't have module level variables)